src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/csip/CSIPService.kt

```diff
@@ -46,24 +49,39 @@ class CSIPService(
       transform {
         dpsApiService.getCsipReport(dpsCsipReportId)
           .let { dpsCsipRecord ->
-            nomisApiService.upsertCsipReport(dpsCsipRecord.toNomisUpsertRequest()).let { nomisCsip ->
-              CSIPFullMappingDto(
-                dpsCSIPReportId = dpsCsipReportId,
-                nomisCSIPReportId = nomisCsip.nomisCSIPReportId,
-                mappingType = CSIPFullMappingDto.MappingType.DPS_CREATED,
-                attendeeMappings = listOf(),
-                factorMappings = listOf(),
-                interviewMappings = listOf(),
-                planMappings = listOf(),
-                reviewMappings = listOf(),
-              )
-            }
+            nomisApiService.upsertCsipReport(dpsCsipRecord.toNomisUpsertRequest())
+              .let {
+                it.createNewMappings(dpsCsipReportId)
+              }
           }
       }
       saveMapping { mappingApiService.createMapping(it) }
     }
   }
 
+  fun UpsertCSIPResponse.createNewMappings(dpsCsipReportId: String) =
```

Comment:
Not sure why I hung this off UpsertCSIPResponse - will change in another PR
All mappings are now passed through to the mapping service for the insert